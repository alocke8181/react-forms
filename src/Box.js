import React from "react";

const Box = (props) =>{
    let {id,bgColor, width, height, removeBox} = props;

    const handleRemove = (e) =>{
        e.preventDefault();
        removeBox(id)
    }

    return (
        <div id={id} className='box' style={{backgroundColor : bgColor, width:`${width}px`, height : `${height}px`}}>
            <button className="box-remove-button" onClick={handleRemove}>X</button>
        </div>
    )

}

export default Box;