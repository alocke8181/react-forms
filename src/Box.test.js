import { render } from '@testing-library/react';
import Box from './Box';

const removeBoxTest = () => console.log('remove')

test('smoke', ()=>{
    render(<Box id={1} bgColor={'#000000'} width={100} height = {100} removeBox={removeBoxTest}/>)
})

test('snapshot',()=>{
    const {asFragment} = render(<Box id={1} bgColor={'#000000'} width={100} height = {100} removeBox={removeBoxTest}/>)
    expect(asFragment()).toMatchSnapshot();
})