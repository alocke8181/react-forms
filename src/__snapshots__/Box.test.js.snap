// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
    id="1"
    style="background-color: rgb(0, 0, 0); width: 100px; height: 100px;"
  >
    <button
      class="box-remove-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
